File: src/main/java/com/oyosite/ticon/specutils/ink/ColorLockedInkStorage.java
package com.oyosite.ticon.specutils.ink
		;

import de.dafuqs.spectrum.api.energy.*;
import de.dafuqs.spectrum.api.energy.color.*;
import de.dafuqs.spectrum.api.energy.storage.*;
import de.dafuqs.spectrum.helpers.*;
import net.minecraft.network.chat.*;

import java.util.*;

public class ColorLockedInkStorage extends SingleInkStorage {

	public ColorLockedInkStorage(long capacity, InkColor color, long amount) {
		super(capacity, color, amount);
	}

	@Override
	public boolean accepts(InkColor color) {
		return this.storedColor == color;
	}

	@Override
	public long addEnergy(InkColor color, long amount) {
		return super.addEnergy(color, amount);
	}

	@Override
	public void setEnergy(Map<InkColor, Long> colors, long total) {
		super.setEnergy(Map.of(this.storedColor, colors.get(this.storedColor)), total);
	}

	@Override
	public void addTooltip(List<Component> tooltip) {
		tooltip.add(Component.translatable("item.spectrum.ink_storage.stores_up_to_ink_per_type", new Object[]{Support.getShortenedNumberString(this.maxEnergy)}));
		if (this.storedEnergy > 0L) {
			InkStorage.addInkStoreBulletTooltip(tooltip, this.storedColor, this.storedEnergy);
		}
	}

}

File: src/main/kotlin/com/oyosite/ticon/specutils/ink/ColorLockedInkStorage.kt
package com.oyosite.ticon.specutils.ink

import de.dafuqs.spectrum.api.energy.InkStorage
import de.dafuqs.spectrum.api.energy.color.InkColor
import de.dafuqs.spectrum.api.energy.storage.SingleInkStorage
import de.dafuqs.spectrum.helpers.Support
import net.minecraft.network.chat.Component
import java.util.Map

class ColorLockedInkStorage(capacity: Long, color: InkColor?, amount: Long) :
    SingleInkStorage(capacity, color, amount) {
    override fun accepts(color: InkColor?): Boolean {
        return this.storedColor === color
    }

    override fun addEnergy(color: InkColor?, amount: Long): Long {
        return super.addEnergy(color, amount)
    }

    override fun setEnergy(colors: MutableMap<InkColor?, Long?>, total: Long) {
        super.setEnergy(Map.of<InkColor?, Long?>(this.storedColor, colors.get(this.storedColor)), total)
    }

    override fun addTooltip(tooltip: MutableList<Component?>) {
        tooltip.add(
            Component.translatable(
                "item.spectrum.ink_storage.stores_up_to_ink_per_type",
                *arrayOf<Any?>(Support.getShortenedNumberString(this.maxEnergy))
            )
        )
        if (this.storedEnergy > 0L) {
            InkStorage.addInkStoreBulletTooltip(tooltip, this.storedColor, this.storedEnergy)
        }
    }
}